qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp

```diff
@@ -195,7 +196,15 @@ void MainWindowPresenter::notifyLoadBatchRequested(int tabIndex) {
   auto filename = QFileDialog::getOpenFileName();
   if (filename == "")
     return;
-  auto map = MantidQt::API::loadJSONFromFile(filename);
+  QMap<QString, QVariant> map;
+  try {
+    map = MantidQt::API::loadJSONFromFile(filename);
+  } catch (const std::runtime_error &e) {
+    QMessageBox msgBox;
```

Comment:
`QMessageBox` has [static functions](https://doc.qt.io/qt-5/qmessagebox.html#warning) that make this simpler.